src/psyclone/psyir/transformations/intrinsics/matmul2code_trans.py

```diff
@@ -478,19 +532,37 @@ def _apply_matrix_matrix(node):
         assign = Assignment.create(result_ref.copy(), rhs)
         # Create ii loop and add the above code as a child
         # Work out the bounds
-        lower_bound, upper_bound, step = _get_array_bound(matrix1, 1)
-        # Must be the same as _get_array_bound(matrix2, 0)
+        if len(m1_fr_order) == 0:
+            # If no full ranges, then matrix was specified
+            # without them i.e.: only has two dimensions.
+            pos_last = 1
+        else:
+            pos_last = m1_fr_order[-1]
+        lower_bound, upper_bound, step = _get_array_bound(matrix1, pos_last)
+        # Must be the same as _get_array_bound(matrix2, pos_first)
         iiloop = Loop.create(ii_loop_sym, lower_bound, upper_bound, step,
                              [assign])
         # Create "result(i,j) = 0.0"
         assign = Assignment.create(result_ref.copy(),
                                    Literal("0.0", REAL_TYPE))
         # Create i loop and add assignment and ii loop as children.
-        lower_bound, upper_bound, step = _get_array_bound(matrix1, 0)
+        if len(m1_fr_order) == 0:
+            # If no full ranges, then matrix was specified
+            # without them i.e.: only has two dimensions.
+            pos_first = 0
+        else:
+            pos_first = m1_fr_order[0]
+        lower_bound, upper_bound, step = _get_array_bound(matrix1, pos_first)
         iloop = Loop.create(i_loop_sym, lower_bound, upper_bound, step,
                             [assign, iiloop])
         # Create j loop and add i loop as child.
-        lower_bound, upper_bound, step = _get_array_bound(matrix2, 1)
+        if len(m2_fr_order) == 0:
+            # If no full ranges, then matrix was specified
```

Comment:
We could get rid of all these "if no full ranges" branches by applying `Reference2ArrayRangeTrans` as part of this transformation but that's a bigger change and this is urgent.